package/oxide/package

```diff
@@ -69,25 +72,39 @@ tarnish() {
     pkgdesc="Service managing power states, connectivity and buttons"
     url=https://github.com/Eeems/oxide/tree/master/applications/system-service
     section=utils
+    depends=(xochitl)
 
     package() {
         install -D -m 644 -t "$pkgdir"/etc/dbus-1/system.d "$srcdir"/etc/dbus-1/system.d/codes.eeems.oxide.conf
         install -D -m 644 -t "$pkgdir"/lib/systemd/system "$srcdir"/etc/systemd/system/tarnish.service
         install -D -m 755 -t "$pkgdir"/opt/bin "$srcdir"/opt/bin/tarnish
     }
-
     configure() {
         systemctl daemon-reload
     }
-
     preremove() {
-        if systemctl list-units --full -all | grep -Fq 'tarnish.service'; then
-            echo "Disabling $pkgname"
-            systemctl disable --now tarnish
+        if is-active tarnish; then
+            echo "Stopping tarnish"
+            systemctl stop tarnish
+        fi
+        if is-enabled tarnish; then
+            echo "Disabling tarnish"
+            systemctl disable tarnish
```

Comment:
`systemctl disable --now tarnish` will stop the unit even if it was not enabled. So I’m not sure whether separating the two actions is actually needed here.

Reply:
Because it might not be enabled so just doing a `is-enabled` check beforehand could miss that `is-active` is true.

Reply:
I see the reasoning. This should also be applied to other packages that declare systemd services then, and maybe factored out int `install-lib`. But let’s keep this PR focused on Oxide.